[WIP] Core concepts rejigger

pdaoust opened the pull request:
- [x] Reorder concepts
    * 8 becomes N2N messaging
    * 9 becomes conflict resolution
- [x] Clean up text in 7 and 8
- [x] Add concepts 7 and 8 to the ToC
- [x] Add concepts 7 and 8 to the intro nav
- [ ] Digitise images for concepts 7 and 8 (B&W only; no colour)
- [x] Convert concepts 1 – 8 to Donna's proposed hybrid style
    - [x] 1: The basics
    - [x] 2: App architecture
    - [x] 3: Source chain
    - [x] 4: DHT
    - [x] 5: Links
    - [x] 6: Crud
    - [x] 7: Validation
    - [x] 8: N2N messaging

freesig commented:
Hey we can't do branch -> master. Please point this at develop

pdaoust commented:
> Hey we can't do branch -> master. Please point this at develop

Crap, I'm sorry. Fat-fingered at the end of a day. Totally meant to do `develop`. Fixing now.

freesig commented:
Looks good to me. Does it look like the divs are working correctly on you're local server?

Just remember to pull develop before merging :)

pdaoust commented:
@freesig yep, the divs and the markdown inside them are working just fine as long as `pymdownx.extra` is enabled.

pdaoust commented:
Okay, I'm up to concept numero 4 now!

pdaoust commented:
@dhtnetwork @freesig @Klayemore requesting review; thank you! There are some fairly substantial structural changes in here. https://affectionate-swartz-15c661.netlify.com/docs/concepts/

freesig commented:
What's the state of this PR. Is it planned for 0.2.3?

dhtnetwork commented:
@freesig correct as noted on the Trello board, it's marked for 0.2.3. Only minor formatting changes in the 0.2.2 release.

pdaoust commented:
@dhtnetwork @freesig @Klayemore I've made some large updates:

* All eight chapters now follow our consistent structure
* Lots of editing to make it work in that structure, simplify language, etc (I still need to go through it and look for ten-dollar words to simplify)
* Shortened titles to one-or-two-word concepts rather than phrases (take a look at what I did on the actual articles though -- I start out with the short phrase then elaborate with a few words. Keep/leave?)
* Moved some diagrams off the DHT page (4) -- the ones pertaining to validation -- and reworked/moved them to the validation page (7).

Live version here: https://5dcdd6bbf70e182ca2aaa516--affectionate-swartz-15c661.netlify.com/docs/concepts/

dhtnetwork commented on src/concepts/index.md:
```diff
@@ -2,15 +2,15 @@ title: Holochain Core Concepts - Holochain Docs
 # Holochain Core Concepts
 
-Welcome to the Holochain Core Concepts! Here we'll introduce you to the basics of Holochain, a framework and network protocol for building secure distributed applications. Holochain is different from what you may be used to, but don't worry&mdash;we'll go at a comfortable pace, building on things you already know.
+Welcome to the Holochain Core Concepts! Here we'll introduce you to the basics of Holochain, a framework and network protocol for building secure distributed applications. Holochain is different from what you may be used to, but don't worry---we'll go at a comfortable pace, building on things you already know.
```
Is it possible to steer away from double negatives? e.g., don't worry, no problem. I think in this case, we can just remove it all together.

pdaoust replied:
which double negative are you referring to? I don't see one in that paragraph.

dhtnetwork replied:
I removed it : ). The double negative is --> don't worry

dhtnetwork commented on src/concepts/7_validation.md:
```diff
@@ -0,0 +1,167 @@
+
+If you're using our [Rust HDK](https://developer.holochain.org/api/latest/hdk/), your validation rule for any entry or link type is simply a function that takes a string, analyzes it, and returns a result (`Ok` or error message).
+
+If your entry holds structured data such as JSON, however, it's annoying to hand-roll your own parsing code so the HDK lets you declare a validation function to accept a [struct](https://doc.rust-lang.org/rust-by-example/custom_types/structs.html) of your own creation. If it sees this sort of function, it'll automatically assume that the entry content is JSON and try to deserialize it into an ins ...
```
Minor - is there a better word than annoying? Otherwise, it's fine.

dhtnetwork commented:
> * All eight chapters now follow our consistent structure  (awesome)
> * Lots of editing to make it work in that structure, simplify language, etc (I still need to go through it and look for ten-dollar words to simplify)  - really nice
> * Shortened titles to one-or-two-word concepts rather than phrases (take a look at what I did on the actual articles though -- I start out with the short phrase then elaborate with a few words. Keep/leave?)  - Keep I like it.  The short phrase + few words look great. 
> * Moved some diagrams off the DHT page (4) -- the ones pertaining to validation -- and reworked/moved them to the validation page (7).  
> 
> Live version here: https://5dcdd6bbf70e182ca2aaa516--affectionate-swartz-15c661.netlify.com/docs/concepts/

We may want to double check em dashes and formatting.  


> * All eight chapters now follow our consistent structure
> * Lots of editing to make it work in that structure, simplify language, etc (I still need to go through it and look for ten-dollar words to simplify)
> * Shortened titles to one-or-two-word concepts rather than phrases (take a look at what I did on the actual articles though -- I start out with the short phrase then elaborate with a few words. Keep/leave?)
> * Moved some diagrams off the DHT page (4) -- the ones pertaining to validation -- and reworked/moved them to the validation page (7).
